fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,527 @@
+            image.save(save_path)
+    elif isinstance(image, PIL.JpegImagePlugin.JpegImageFile):
+        save_path = os.path.join(path, basename + ext)
+        if not os.path.exists(save_path):
+            image.save(save_path)
+    elif isinstance(
+        image,
+        (PIL.MpoImagePlugin.MpoImageFile, PIL.WebPImagePlugin.WebPImageFile),
+    ):
+        save_path = os.path.join(path, basename + ext)
+        if os.path.exists(save_path):
+            return save_path
+        image_bytes = io.BytesIO()
+        image.save(image_bytes, format="JPEG")
+        image_bytes.seek(0)
+
+        with open(save_path, "wb") as file:
+            file.write(image_bytes.getvalue())
+    elif isinstance(image, PIL.GifImagePlugin.GifImageFile):
+        save_path = os.path.join(path, basename + ext)
+        if os.path.exists(save_path):
+            return save_path
+        image.convert("RGB").save(save_path, "JPEG")
+    else:
+        raise ValueError(f"Image type {type(image)} not supported")
+
+    return save_path
+
+
+def _get_download_dir(repo_id, *hf_args):
+    download_dir = os.path.join(
+        fo.config.default_dataset_dir, "huggingface", "hub", repo_id
+    )
+    if len(hf_args) > 0:
+        download_dir = os.path.join(download_dir, *hf_args)
+
+    if not os.path.exists(download_dir):
+        os.makedirs(download_dir)
+
+    return download_dir
+
+
+def _get_image_field_name(hf_dataset):
+    if isinstance(hf_dataset, datasets.DatasetDict):
+        split_names = list(hf_dataset.keys())
+        return _get_image_field_name(hf_dataset[split_names[0]])
+
+    for feature_name, feature in hf_dataset.features.items():
+        if isinstance(feature, datasets.features.Image):
+            return feature_name
+    return None
+
+
+def _add_images(
+    fo_dataset, hf_dataset, image_field, download_dir, split_name=None
+):
+    samples = []
+    for i, item in enumerate(hf_dataset):
+        image = item[image_field]
+        basename = f"{image_field}_{i:08d}"
+        save_path = _save_PIL_image_to_disk(image, download_dir, basename)
+        sample_dict = {"filepath": save_path}
+        if split_name is not None:
+            sample_dict["tags"] = [split_name]
+        samples.append(fo.Sample(**sample_dict))
+    fo_dataset.add_samples(samples)
+
+
+def _convert_hf_value_dtype(hf_value):
+    dtype = hf_value.dtype
+    if dtype in ["int8", "int16", "int32", "int64"]:
+        return fo.IntField
+    elif dtype in ["float16", "float32", "float64"]:
+        return fo.FloatField
+    elif dtype == "bool":
+        return fo.BooleanField
+    elif dtype == "string":
+        return fo.StringField
+    else:
+        raise ValueError(f"Unknown dtype {dtype}")
+
+
+def _is_classification_field(feature):
+    return isinstance(feature, datasets.features.ClassLabel)
+
+
+def _add_classification_field(fo_dataset, hf_dataset, feature_name):
+    print(f"Adding classification field {feature_name} to dataset")
```

Comment:
reminder to remove `print()` statements before merging, unless you are intentionally including them, in which case our convention is to use the `logging` module as follows so users can configure whether/how logging is done:

``py
logger = logging.getLogger(__name__)

logger.info("your message")
```

Reply:
Got it. Didn't realize this, will do from now on